
In this exercise, you will practice basic `dplyr` (the relevant package within `tidyverse`) operations on a dataset of gene expression levels across human tissues (from the GTEx Project). The dataset (`dt`) has one row per gene, with columns:

-   `Name`: the Ensembl gene ID
-   `Description`: the gene symbol
-   One column per tissue containing the median expression levels (in TPM units)

------------------------------------------------------------------------

### Start by loading the data into R:

Note the arguments, which specify that the first two lines of the file will be skipped, that the file contains a header line that should be used to form column names, and that the columns are tab (`\t`) separated.

```{r}
library(tidyverse)
getwd()
setwd("/Users/cmdb/Desktop/bootcamp")
dt <- read_delim("GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_median_tpm.gct.gz", skip = 2, col_names = TRUE, delim = '\t')
```

### Q1. How many genes are in the dataset, and how many tissues are represented?

```{r}
 # code and answer here

nrow(dt)  #number of genes is 56200
ncol(dt) - 2 #number of tissues is 54
```

------------------------------------------------------------------------

### Q2. How many genes are expressed above a given threshold in each tissue?

A common threshold is 1 TPM to define whether a gene is "expressed."\
Compute, for each of the three tissues, `Liver`, `Testis`, and `Whole Blood`, how many genes are expressed above this threshold.\
Then identify which tissue has the **most** and which has the **fewest** expressed genes. See this paper for one possible explanation: Xia et al. 2021: [PMC7891839](https://pmc.ncbi.nlm.nih.gov/articles/PMC7891839/).

Do the results change if you set the threshold higher, say at 10 TPM?

```{r}

# Code here
Liver <- dt%>%select(Description,Liver)%>%filter(Liver>1)
Testis <- dt%>%select(Description,Testis)%>%filter(Testis>1)
WholeBlood <- dt%>%select(Description,`Whole Blood`)%>%filter(`Whole Blood`>1)


Liver10 <- dt%>%select(Description,Liver)%>%filter(Liver>10)
Testis10 <- dt%>%select(Description,Testis)%>%filter(Testis>10)
WholeBlood10 <- dt%>%select(Description,`Whole Blood`)%>%filter(`Whole Blood`>10)

# Interpretation here as a comment
#Genes expressed with a 1 TPM threshold are the following : Liver (13499), Testis (25064), Whole Blood (11616)
#Genes expressed with a 10 TPM threshold are the following : Liver (5132), Testis (11471), Whole Blood (4482)
#When the threshold is greater than 1PM, out of the three tissues, the Testis (25064) has the greatest expressed genes while the wholeblood (11616) has the lowest expressed genes. 
#When the threshold is greater than 10PM, out of the three tissues, the Testis (11471) has the greatest expressed genes while the wholeblood has the lowest expressed genes (4482). Therefore, the results do not change.


```

------------------------------------------------------------------------

### Q3. What are the 20 most highly expressed genes in the liver?

What do you notice about many of the genes in this list? Can you propose a hypothesis to explain your observation?

```{r}

# Code here
dt%>%arrange(desc(Liver))%>%select(Name,Description, Liver)%>%head(Liver, n = 20)

# Interpretation here as a comment
#The genes in this list are mostly mitochondrial DNA. This could possibly be explained by the Liver's function and importance in metabolic processes, and mitochondrial DNA role in producing relevant proteins for energy processes in the cell.  


```

------------------------------------------------------------------------

### Q4. Compare the mean expression levels of mitochondrial versus nuclear-encoded genes in the Liver.

Genes encoded by the mitochondrial genome have the gene symbol prefix `MT-`. The function `str_starts()` from the `stringr` package in `tidyverse` tests whether a string (or a vector of strings) start with a given pattern, returning TRUE/FALSE (or a boolean vector) as output.

Use `str_starts()` to identify which genes are mitochondrially encoded and store this information in a new Boolean column named `is_mitochondrial`. Next, compute the mean and median expression of mitochondrial genes in the liver, as well as the mean and median expression of nuclear-encoded genes in the liver and compare them. Do mitochondrial genes exhibit higher or lower mean and median expression than nuclear genes? What does the difference between the mean and median tell you about the distribution of gene expression.

```{r}
# Code here

#groupby works on columns not the row values 
dt_exp <- dt %>%
  mutate(is_mitochondrial = str_starts(Description,"MT-")) %>%
  group_by(is_mitochondrial) %>% 
  summarise(meanEx = mean(Liver, na.rm = TRUE))

dt_expmedian <- dt %>%
  mutate(is_mitochondrial = str_starts(Description,"MT-")) %>%
  group_by(is_mitochondrial) %>% 
  summarise(median = median(Liver, na.rm = TRUE))

dt_exp
dt_expmedian
# Interpretation here as a comment
#Mitochondrial genes exhibit higher mean, 11592.716143, than nuclear-encoded genes,8.373839, in the liver. Mitochondrial genes exhibit higher median, 3.40258, than nuclear-encoded genes,0, in the liver. This indicates that mitochondrial genes are very active in comparison to nuclear-encoded genes

```

------------------------------------------------------------------------

### Q5. Are liverâ€™s most highly expressed genes also highly expressed in another tissue?

Take the 20 most highly exprssed genes in **Liver** and check how they are expressed in the **Heart - Left Ventricle**. Identify:\
1. one gene that is highly expressed in both tissues\
2. one gene that is highly expressed in Liver but very low in Heart

Look up basic information about these genes on the internet. Do their expression patterns make sense given their known functions?

```{r}

# Code here
dt%>%select(Description,Liver,`Heart - Left Ventricle`)%>%arrange(desc(Liver))%>%head(Liver, n = 20)%>%mutate

# Interpretation here as a comment
#One gene that is highly expressed in both tissues is MT-ND4, 58775.30(liver)	, 	6.60157e+04	(heart)	
#One gene that is highly expressed in the Liver but lower in the heart is FGB, in the liver its expressed #at 7351.90	while the heart it's expressed at 2.85234e-02	
#The MT-ND4 gene is important for the electron transport chain which is important for energy production in the body, so it makes sense as to why it's highly expressed in important tissues such as the Lung and Heart. 
#The FGB gene plays an important role in blood coagulation, and the liver plays a role in producing clotting factors. However, in the heart, having high expression of genes that causes clotting is dangerous as it can lead to diseases or heart attacks. 

```

------------------------------------------------------------------------

### Q6. Comparing gene expression between tissues

Another way to compare tissues is to ask how similar are their expression patterns overall, across all genes?

R has a built-in function called `cor.test()` that measures the correlation between two vectors of numbers. Remember that you can extract the contents of a given column of a tibble as a vector using the syntax `tibble_name$column_name`. If a column name contains spaces, enclose the column name in backticks such as `` tibble_name$`column name` ``.

Use `cor.test()` with the argument xc to compute: - the Spearman correlation between `Liver` and `Heart - Left Ventricle` - the Spearman correlation between `Liver` and `Brian - Cortex`

Which pair of tissues is more strongly correlated? Does this result make sense biologically?

```{r}
# Liver vs Heart answer
#p-value < 2.2e-16, z = 241.7
cor.test(dt$Liver, dt$`Heart - Left Ventricle`, method = "kendall")
# Liver vs Brain answer
#z = 223.7, p-value < 2.2e-16
cor.test(dt$Liver, dt$`Brain - Cortex`, method = "kendall")
# interpretation

# The Liver and Heart are more strongly correlated since it has a z value of 241.7. This results make biological sense as both of these tissues play an important role in circulating blood and other physiological processes. 

```

------------------------------------------------------------------------

### Optional: further exploration

What other patterns are present in the data? Use tidyverse functions to explore and summarize the data in different ways and see what you can learn.

```{r}
# answer here
```
